build: improve roachtest build scripts

RaduBerinde opened the pull request.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/111251)
<!-- Reviewable:end -->

RaduBerinde committed:
build: improve roachtest build scripts

We improve the roachtest scripts to support building finer-grained
components: the `roachtest_compile_component.sh` script builds a
single component (cockroach, cockroach-ea, workload, libgeos,
roachtest).

In addition, the script now recognizes a `ROACHTEST_PREBUILT_BINARIES`
env var; if it is set to 1, it just checks that the necessary build
products are already present.

With these changes, we can set up separate teamcity builds for
producing the binaries and we can use the pre-built binaries in the
roachtest build.

Epic: none
Release note: None
